lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -124,11 +112,13 @@ public function getExtension()
      */
     protected function getFilename($id)
     {
+        $hash = hash('sha256', $id);
```

Comment:
I implemented another versiong using bin2hex by default and fallback to the hash if id would be too long. Also I fixed the naming if id is empty, where bin2hex wouldn't work as well.